src/Catalyst.Protocol/Transaction/PublicEntry.cs

```diff
@@ -33,13 +34,52 @@ public partial class PublicEntry
 
         public bool IsValid()
         {
-            if (Amount == null || Amount.IsEmpty || Amount.All(b => b == default))
+            // make it a constant - MIN_GAS_LIMIT
+            if (GasLimit < 21000)
+            {
+                return false;
+            }
+
+            var isTimestampValid = Timestamp != default(Timestamp) && Timestamp != new Timestamp();
+            if (!isTimestampValid)
             {
-                Logger.Debug("{field} cannot be 0", nameof(Amount));
+                Logger.Debug("{timestamp} cannot be null or 0.");
                 return false;
             }
 
-            return true;
+            var hasValidSignature = Signature.IsValid(SignatureType.TransactionPublic);
+            return hasValidSignature;
         }
+
+        // // add to proto
+        // /// <summary>
+        // /// Gas limit for the entry expressed in gas units.
+        // /// </summary>
+        // public ulong GasLimit { get; set; }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4600656)